toonz/sources/image/ffmpeg/tiio_gif.cpp

```diff
@@ -0,0 +1,261 @@
+
+#include "tsystem.h"
+#include "tiio_gif.h"
+#include "trasterimage.h"
+#include "timageinfo.h"
+#include <QStringList>
+
+
+//===========================================================
+//
+//  TImageWriterGif
+//
+//===========================================================
+
+class TImageWriterGif : public TImageWriter {
+public:
+	int m_frameIndex;
+
+	TImageWriterGif(const TFilePath &path, int frameIndex, TLevelWriterGif *lwg)
+		: TImageWriter(path), m_frameIndex(frameIndex), m_lwg(lwg) {
+		m_lwg->addRef();
+		
+	}
+	~TImageWriterGif() { m_lwg->release(); }
+
+	bool is64bitOutputSupported() override { return false; }
+	void save(const TImageP &img) override { m_lwg->save(img, m_frameIndex); }
+
+private:
+	TLevelWriterGif *m_lwg;
+
+};
+
+
+//===========================================================
+//
+//  TLevelWriterGif;
+//
+//===========================================================
+
+TLevelWriterGif::TLevelWriterGif(const TFilePath &path, TPropertyGroup *winfo)
+	: TLevelWriter(path, winfo) {
+	if (!m_properties) m_properties = new Tiio::GifWriterProperties();
+	std::string scale = m_properties->getProperty("Scale")->getValueAsString();
+	sscanf(scale.c_str(), "%d", &m_scale);
```

Comment:
You can just use `std::atoi()` or `std::strtol()` from `<cstdlib>` instead when you want to do string to int conversion. Using a format string and `std::sscanf()` is slightly overkill.

Not trying to criticize you at all, just thought I'd point it out since you said you were new to C++.